### PR TITLE
gitindex: index symbolic links

### PR DESCRIPTION
**Pull request by stefanhengl**
Up to now we skipped symbolic links when compiling the list of files
during indexing. With this change we index symbolic links with the
relative path of their target file as their content.

### CODE REVIEW

File: gitindex/tree.go [hunk 1]
```diff
@@ -178,7 +178,7 @@ func (r *repoWalker) handleEntry(p string, e *object.TreeEntry) error {
 	}
 
 	switch e.Mode {
-	case filemode.Regular, filemode.Executable:
+	case filemode.Regular, filemode.Executable, filemode.Symlink:
```
Comment (tsenart) [inline gitindex/tree.go:181]: Would we support symlinked directories with this change?

**Review by tsenart — APPROVED** [review]

File: gitindex/tree.go (hunk 1 above)
Comment (stefanhengl) [reply gitindex/tree.go:181]: Yes. Symlinked directories created with `ln -s target_dir link` are just symbolic links. They are stored with the same mode bits as a symlinked file in the object store trees. Their blobs contain the relative paths. From perspective of Zoekt there is no difference.

**Comment by stefanhengl** [comment]
@keegancsmith Based on a customer request we are adding support to index symlinks. I was surprised to see that we didn't support this before, so I am wondering whether there is a reason not to do it? Note that we don't index the content of the target file but only the relative path that points to it.

**Review by keegancsmith — APPROVED** [review]
LGTM. Indexing the content of the symlink and not the target sounds good to me. Should we also see what unindexed search does and make sure we have the same behaviour? I assume it will be a similiar small change?

**Pull request merged**